### ISSUE TITLE
(ASDK 1.9.3) ASTextNode truncates on first line instead of second line for single words that are long enough

### ISSUE THREAD

**Issue by samhsiung** [facebook/AsyncDisplayKit#930]
For "hsiung@gmail.com" (see the last icon and caption in the screenshot below) it renders the text in two lines without truncation. However a longer email address such as "hsiung@mailinator.com" will result in truncation on the first line (see the first icon and caption in the screenshot below). In 1.9.2 ASTextNode was truncating correctly on the second line.

![simulator screen shot dec 9 2015 11 09 16 am](https://cloud.githubusercontent.com/assets/194822/11700023/32c1c59e-9e7c-11e5-8c6a-f48292f31a7b.png)

Text node:
```
ASTextNode *textNode = NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
        paragraphStyle.lineBreakMode = NSLineBreakByWordWrapping;
        paragraphStyle.paragraphSpacing = paragraphStyle.lineSpacing;
        paragraphStyle.alignment = NSTextAlignmentCenter;
        UIFont *font = [UIFont fontWithName:@"HelveticaNeue-Bold" size:12.0];
        NSMutableDictionary *attributes = [[NSMutableDictionary alloc] initWithDictionary:@{ NSParagraphStyleAttributeName : paragraphStyle,
                                                                                             NSFontAttributeName : font,
                                                                                             NSForegroundColorAttributeName : [UIColor blackColor],
                                                                                             NSKernAttributeName : @(-0.2)}];
        [textNode setAttributedString:[[NSAttributedString alloc] initWithString:@"averylongemail@address.com" attributes:attributes]];
```
Layout code:
```
    CGSize exactSize = CGSizeMake(60, 30);
    ASStackLayoutSpec *verticalStack = [ASStackLayoutSpec verticalStackLayoutSpec];
    verticalStack.alignItems = ASStackLayoutAlignItemsStretch;
    verticalStack.children = @[textNode];
    verticalStack.layoutPosition = CGPointZero;
    verticalStack.sizeRange = ASRelativeSizeRangeMakeWithExactCGSize(exactSize);
    ASStaticLayoutSpec *staticSpec = [[ASStaticLayoutSpec alloc] init];
    staticSpec.children = @[verticalStack];
```

**Comment by appleguy** [comment]
@ocrickard @levi this is a regression from the CK text engine...any ideas?

**Comment by ocrickard** [comment]
I won't have time to look into this for the next week or so.

**Comment by levi** [comment]
Damn, this is unfortunate. @samhsiung do you have time to write a snapshot test to reproduce this issue? I can take a deeper look when we've got a solid test case available. Life is a little overwhelming for me at the moment, so won't be able to look at this until later in the weekend.

**Comment by samhsiung** [comment]
No problem @levi, also swamped at the moment with other things at work, but I can take a look during the weekend.

**Comment by samhsiung** [comment]
Was able to get it to work by setting 'useFontLeading = YES' for ASTextKitContext's layoutManager. We are customizing the leading for our fonts so there was a discrepancy between the layout+truncation calculation and actual rendering

**Comment by samhsiung** [comment]
Actually we're not customizing font leading but the linespacing which is at the paragraph level. Setting paragraph line spacing = 1 breaks ASTextNodeTests.m so something else must be going on.

**Comment by samhsiung** [comment]
This should be fixed in #941. The truncation calculation for the last line wasn't taking attributes from the set attributedString into account.
